Automattic/woocommerce-payments
# Added filter to update Level 3 data.

#### Changes proposed in this Pull Request


##### Allow filtering level 3 data

This PR adds support to allow the use of WooCommerce gift cards to pay for order partially with full support for Level 3 data, with the rest being paid by another payment method provided by Woo Payments. This previously failed due to Level 3 data mismatch.



##### Description

Together with https://github.com/woocommerce/woocommerce-gift-cards/pull/822 serves as a POC to show how Gift Cards can be used as discounts with Woo Payments. 

More background discussion available at p1718202231605999-slack-C3NCP7ZJ6

Initially, I only added the filter, but after working a bit more on the change in Gift Cards, it seems to me it's better if the payment gateway handles the bundling of line items after filtering for the following reasons:
- WooPayments has more control over the validity of the request to Stripe
- Users of this filter don't have to implement their own checking for limit in number of line items
- If the limit changes in the future, it would be difficult to trace all usages/copies of the bundling code to update it. Now it's centralized in an extension that is closer to Stripe API (compared to Gift Cards, where nobody pays attention to changes in Stripe API)

On the other hand, it provides slightly less flexibility to the extension, but that would be an illusion of flexibility anyway, since the ability to send an invalid request to Stripe is worthless.

A similar chance is being done also in Stripe: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3286





#### Testing instructions





* Please check [the GC PR](https://github.com/woocommerce/woocommerce-gift-cards/pull/822).

-------------------


**Post merge**
